### ISSUE TITLE
More flexible variable assignment

### ISSUE THREAD

**Issue by jasal82** [conan-io/conan#5309]
The variable `env_info.path` in profiles supports two types of assignment semantics. If set to a string it overwrites the existing variable. However, if you set it to an array it appends to PATH. I think that this behaviour is needed for all other variables as well. Otherwise it is impossible to create environment layers that can be reused properly.

Profiles have the include command which can be used to import another profile. However, environment settings in a single variable can not be split up between these profiles because the last one always wins with the current assignment semantics. A use-case for this would be a basic toolchain profile which defines architecture settings in the compiler flags and is included in antoher, more specific profile, which adds processor-specific settings to the compiler flags. Currently I see no way of implementing this.

Ideally I'd like to have something similar to the variable assignment operators that can be used in Yocto recipes. See https://www.yoctoproject.org/docs/2.7/mega-manual/mega-manual.html#setting-a-default-value and the following sections for reference.

**Comment by lasote** [comment]
To confirm, are the env vars in profiles overwritten (when including other profiles) even if declared as:

```
[env]
PATH=[/some/path/to/my/tool]
```

?

**Comment by jasal82** [comment]
The array pattern can be used for other variables than `PATH` as well, but they are joined with a colon which doesn't make sense for compiler flags for example.

**Comment by lasote** [comment]
Yes, that is already reported here and also an issue in the conanfiles: https://github.com/conan-io/conan/issues/3853

**Comment by jasal82** [comment]
Regarding the variable assignment and expansion mechanism I see two more or less elegant ways of implementing this:

1. Overload operators to get something similar to the Yocto approach. Could be a bit confusing though.
2. Assign objects to the variables that can be lazily evaluated. Here is an example:

```
self.env_info.path = assignIfEmpty(["path1", "path2", "path3"])
self.env_info.cflags = prepend(["-ffunction-sections", "-fdata-sections"])
```

**Comment by memsharded** [comment]
A whole new model for environment definition has been introduced from Conan 1.40, and this will be the one to move forward in 2.0. Check: https://docs.conan.io/en/latest/reference/conanfile/tools/env.html.

I think this can be closed now, thanks!
